templates/singleton/boil_types.go.tpl

```diff
@@ -60,29 +60,81 @@ MySQL output looks like:    TableNameColNameEnumValue = "enumvalue"
 
 It only titlecases the EnumValue portion if it's snake-cased.
 */}}
+
 {{$once := onceNew}}
 {{- range $table := .Tables -}}
 	{{- range $col := $table.Columns | filterColumnsByEnum -}}
 		{{- $name := parseEnumName $col.DBType -}}
 		{{- $vals := parseEnumVals $col.DBType -}}
 		{{- $isNamed := ne (len $name) 0}}
+        {{- $enumName := "" -}}
 		{{- if and $isNamed (onceHas $once $name) -}}
 		{{- else -}}
 			{{- if $isNamed -}}
 				{{$_ := oncePut $once $name}}
 			{{- end -}}
 {{- if and (gt (len $vals) 0) (isEnumNormal $vals)}}
-// Enum values for {{if $isNamed}}{{$name}}{{else}}{{$table.Name}}.{{$col.Name}}{{end}}
+
+{{- if $isNamed -}}
+    {{ $enumName = titleCase $name}}
+{{- else -}}
+    {{ $enumName = titleCase $table.Name $col.Name}}
+{{- end -}}
+
+{{if $.AddEnumTypes}}
+type {{$enumName}} string
+{{end}}
+
+// Enum values for {{$enumName}}
 const (
-	{{- range $val := $vals -}}
```

Comment:
By unindenting this you're changing the scheme of how these templates are done. This belongs to the `if` statement above on line 76 and so it should remain indented.

The reason for this is because the templates are REAL hairy (as you probably noticed haha) and so keeping the actual template lines indented to match the nesting level helps to keep them a bit cleaner.